singer_sdk/sinks/sql.py

```diff
@@ -144,6 +145,37 @@ def conform_name(self, name: str, object_type: Optional[str] = None) -> str:
         # remove leading digits
         return name.lstrip(digits)
 
+    @staticmethod
+    def _check_conformed_names_not_duplicated(
+        conformed_property_names: Dict[str, str]
+    ) -> None:
+        """Check if conformed names produce duplicate keys.
+
+        Raises:
+            ConformedNameClashException if duplicates found.
+        """
+        duplicates = {
+            x
+            for x in conformed_property_names.values()
+            if list(conformed_property_names.values()).count(x) > 1
+        }
+        if duplicates:
+            duplicated_properties = [
+                key
+                for key, value in conformed_property_names.items()
+                if value in duplicates
+            ]
+            duplicated_property_string = ", ".join(
+                [
+                    f"{dup} -> {conformed_property_names[dup]}"
+                    for dup in duplicated_properties
+                ]
+            )
+            raise ConformedNameClashException(
+                "Duplicate stream properties produced when conforming property names: "
+                + duplicated_property_string
+            )
```

Comment:
There's quite a few loops here. Over

1. dict.values()
2. list(dict.values())
3. list.count()
4. conformed_property_names.items()
5. duplicated_properties

It might be worth simplifying to 2 (or 3 if pretty formatting the dups is a requirement):

```suggestion
        """Check if conformed names produce duplicate keys.

        Raises:
            ConformedNameClashException if duplicates found.
        """
        # group: {'_a': ['1_a'], 'abc': ['aBc', 'abC']}
        grouped = defaultdict(list)
        for k, v in mapping.items():
            grouped[v].append(k)

        # filter
        duplicates = list(filter(lambda p: len(p[1]) > 1, grouped.items()))

        if duplicates:
            raise ConformedNameClashException(
                f"Duplicate stream properties produced when conforming property names: {duplicates}"
            )
```

Reply:
Way better, thanks 🙏